app/src/main/kotlin/org/fossify/paint/views/MyCanvas.kt

```diff
@@ -396,6 +403,10 @@ class MyCanvas(context: Context, attrs: AttributeSet) : View(context, attrs) {
         mListener?.toggleRedoVisibility(visible)
     }
 
+    private fun updateClearConfirmation(edited: Boolean) {
+        mListener?.toggleClearConfirmation(edited)
+    }
```

Comment:
Instead of passing a boolean manually, we can use an in-place check for better readability:

```suggestion
    private fun updateClearConfirmation() {
    	val hasContent = mBackgroundBitmap != null || mOperations.isNotEmpty()
        mListener?.toggleHasContent(hasContent)
    }
```

Reply:
This value doesn't rely only on undo stack. Changing it this way will break how it's working. After clearing, I explicitly set false, despite that undo is available.

Reply:
The new condition seems to work fine for me. Can you provide any example where it won't?

The `updateClearConfirmation(mOperations.isNotEmpty() || mLastOperations.isNotEmpty())` call makes it behave inconsistently compared to how it works when I draw one stroke ➜ undo ➜ clear canvas (no confirmation because canvas was already empty). 

One other case I can think of is loading an image, in which case, `updateClearConfirmation()` should be called _after_ the image is loaded anyway (inside `runOnUiThread`), so we avoid asking for confirmation in case the image fails to load.